ntoskrnl/mm/section.c

```diff
@@ -1329,6 +1410,29 @@ MmNotPresentFaultSectionView(PMMSUPPORT AddressSpace,
                           &MemoryArea->Data.SectionData.RegionListHead,
                           Address, NULL);
     ASSERT(Region != NULL);
+
+    /* Check for a NOACCESS mapping */
+    if (Region->Protect & PAGE_NOACCESS)
+    {
+    	return STATUS_ACCESS_VIOLATION;
```

Comment:
Indentation is pretty broken in this whole hunk.

Reply:
Hmmm yeah. I will have to check the settings in my editor

Reply:
Now this should be fine.